Changelog 1.4.0 (#3328)
btcpayserver/btcpayserver: Changelog.md

# Changelog

## 1.4.0

> This release, while bringing a few new feature and bug fixes, focused on a deep rework of the UI/UX of BTCPay Server.

Reviewer: ```suggestion
This release, while bringing a few new features and bug fixes, focused on a deep rework of the UI/UX of BTCPay Server.
```

This has been mainly the work of @dennisreimann and @dstrukt and the help of many testers.
We can't enumerate the changes that happened here, so see this blog post for a tour of the new UX.

As BTCPay Server started in 2017, we never took the chance to cleanup 4 years of incremental changes.
So we did also did a lot of code refactoring that we can't simply enumerate in the changelog:
* Migrating from .NET 3.1 to .NET 6.0, 
* Use modern C# 10.0 features instead of older constructs
* Class and route renaming

### New features:

> * Greenfield: Add a `missingPermission` field to 403 errors. @NicolasDorier @woutersamaey

Reviewer 2: ```suggestion
* Greenfield: Add a `missingPermission` field to 403 errors (#3195) @NicolasDorier @woutersamaey
```

* Support for new TLS version of SMTP server (#3202) @NicolasDorier
* Greenfield: Added field "StoreId" to a Payment Request (#3223) @woutersamaey
* Greenfield: Can create a payment request without specifying currency (would then use store's default currency) (#3222) @NicolasDorier

### Bug fixes:

* Fix: BTCPay would crash if running in an unexpected working directory (#1894 #3295) @NicolasDorier
* Fix: Can't add security device on safari (#3197 #3322) @Kukks
* If a root path was used, the Notification dropdown wouldn't automatically fetch new notifications @NicolasDorier

> * Clipboard wasn't working over http (mainly in at home setups) @dennisreimann

Reviewer 2: ```suggestion
* Clipboard wasn't working over http, mainly used in at home setups (#3296) @dennisreimann
```

> * Cannot batch unarchive invoices @dennisreimann

Reviewer 2: ```suggestion
* Cannot batch unarchive invoices (#3264) @dennisreimann
```

* Greenfield: Creating a payment request would fail if expiry was specified (#3222) @NicolasDorier
* In wallet's receive if you copy a p2sh address, it would be truncated (#3218) @dennisreimann
* Shopify: Fix partial payments. Generate an invoice based on outstanding amount instead of total. (#3193 #3203) @Kukks
* Btcpayserver instance sends 2 emails after invoice is set as expired, paid or confirmed/complete (#968) @NicolasDorier
* Greenfield: Payment Method update was impossible if using internal ln node while being guest (#2860) @NicolasDorier
* Checkout: Error when changing payment method in invoice (#3075) @dennisreimann
* Greenfield: `created` field of payment request should be a unix timestamp @woutersamaey (#3221)
* Fix LN Node availability check (#3189) @dennisreimann

### Improvements:

* Use the invoice terminology Processing/Settled in the UI rather than Paid/Confirmed/Complete.
* Do not use uppercase in urls (#921) @NicolasDorier

> * Add a copy tor url in the footer (#2692 #3290) @dennisreimann

Reviewer 2: ```suggestion
* Add a copy Tor URL in the footer (#2692 #3290) @dennisreimann
```

# Changelog

## 1.4.0

BTCPay Server started in August 2017 and meanwhile has been evolving incrementally thanks to the feedback of the community.

It was finally time to cleanup the UI/UX and technical debt we accumulated over the years.

> We enumerate here a lot's of new features and bug fixes, but we do not enumerate the UI/UX changes, as it is better experienced in this blog post and videa than explained.

Reviewer 2: ```suggestion
We enumerate here a lot of new features and bug fixes, but we do not enumerate the UI/UX changes, as it is better experienced in this blog post and video than explained.
```

> The heavy lifting of this work has been mainly brought to you thanks to the collaboration of @dstrut and @dennisreimann.

Reviewer 2: ```suggestion
The heavy lifting of this work has been mainly brought to you thanks to the collaboration of @dstrukt and @dennisreimann.
```

We thank also all the testers we brought us feedback, and all of you who participated in the weekly design meetings.

> The work on the UI/UX is however never over, we will incrementally improve it based on your feedback.

Reviewer 2: ```suggestion
The work on the UI/UX is however never over and we will keep on improving it based on your feedback.
```

> Note: If you are using our docker deployment on a raspberry pi 4, there is a small chance your docker version does not support the new docker image.

Reviewer 2: ```suggestion
Note: If you are using our Docker deployment on a Raspberry Pi 4, there is a small chance your Docker version does not support the new docker image.
```

If you have any issue with raspberry pi 4, you need to update your docker version following steps on [this blog post](https://blog.samcater.com/fix-workaround-rpi4-docker-libseccomp2-docker-20/).

> Note that you do not need to update libseccomp2, our udpate process do this for you automatically.

Reviewer 2: ```suggestion
Note that you do not need to update libseccomp2, our update process does this for you automatically.
```

### New features:

* Greenfield: Add a `missingPermission` field to 403 errors (#3195) @NicolasDorier @woutersamaey
* Support for new TLS version of SMTP server (#3202) @NicolasDorier
* Greenfield: Added field "StoreId" to a Payment Request (#3223) @woutersamaey
* Greenfield: Can create a payment request without specifying currency (would then use store's default currency) (#3222) @NicolasDorier
* Add login code, for easy login to BTCPay Server via a mobile device (#2504) @Kukks
* Add LNUrl Auth support as second factor auth (#3083) @Kukks

### Bug fixes:

* Fix: BTCPay would crash if running in an unexpected working directory (#1894 #3295) @NicolasDorier
* Fix: Can't add security device on safari (#3197 #3322) @Kukks
* If a root path was used, the Notification dropdown wouldn't automatically fetch new notifications @NicolasDorier
* Clipboard wasn't working over http, mainly used in at home setups (#3296) @dennisreimann

> * Cannot batch unarchive invoices (#3264) @dennisreimann

Reviewer 2: This should be in the "New features" section as "Batch unarchive invoices"

* Greenfield: Creating a payment request would fail if expiry was specified (#3222) @NicolasDorier
* In wallet's receive if you copy a p2sh address, it would be truncated (#3218) @dennisreimann
* Shopify: Fix partial payments. Generate an invoice based on outstanding amount instead of total. (#3193 #3203) @Kukks

> * Btcpayserver instance sends 2 emails after invoice is set as expired, paid or confirmed/complete (#968) @NicolasDorier

Reviewer 2: ```suggestion
* BTCPay Server instance sends 2 emails after invoice is set as expired, paid or confirmed/complete (#968) @NicolasDorier
```
